stdlib/public/core/StringBridge.swift

```diff
@@ -322,12 +322,23 @@ extension String {
   public // SPI(Foundation)
   func _bridgeToObjectiveCImpl() -> AnyObject {
     if _guts.isSmall {
-      return _guts.asSmall.withUTF8 { bufPtr in
-        return _createCFString(
+      //ASCII Strings a) can bridge to tagged pointers, b) cannot contain a BOM
+      if _guts.isASCII {
+        return _guts.asSmall.withUTF8 { bufPtr in
+          return _createCFString(
             bufPtr.baseAddress._unsafelyUnwrappedUnchecked,
             bufPtr.count,
             kCFStringEncodingUTF8
-        )
+          )
+        }
+      } else {
+        //We can't form a tagged pointer String, so grow to a non-small String,
+        //and bridge that instead. Also avoids CF deleting any BOM that may be
+        //present
+        var copy = self
+        copy._guts.reserveCapacity(_SmallString.capacity + 1)
+        assert(!copy._guts.isSmall)
+        return copy._bridgeToObjectiveCImpl()
       }
     }
     if _guts._object.isImmortal {
```

Comment:
I think the logic would flow a little better as:

```swift
if _guts.isSmallASCII {
  return ...
}
if _guts.isSmall {
  return ...
}
...
```

Reply:
Aha, didn't realize that existed